src/odin/ingestion/qlik/utils.py

```diff
@@ -0,0 +1,270 @@
+    """
+    change_table = f"{table}__ct/"
+    prefixes = (
+        os.path.join(DATA_ARCHIVE, IN_QLIK_PREFIX, change_table),
+        os.path.join(DATA_ERROR, IN_QLIK_PREFIX, change_table),
+    )
+    paths = []
+    for prefix in prefixes:
+        in_func = None
+        if save_local:
+            try:
+                ds = ds_from_path(os.path.join(DATA_SPRINGBOARD, CUBIC_QLIK_DATA, table))
+                filter = pc.starts_with(pc.field("header__from_csv"), f"s3://{prefix}")
+                _, min_path = ds_column_min_max(ds, "header__from_csv", ds_filter=filter)
+            except Exception as _:
+                pass
+            else:
+                if min_path is not None:
+
+                    def filter_func(obj: S3Object) -> bool:
+                        return obj.path > min_path
+
+                    in_func = filter_func
+
+        paths += list_objects(
+            prefix, in_filter=".csv.gz", max_objects=max_change_files, in_func=in_func
+        )
+
+    return paths
+
+
+def snapshot_to_parquet(obj_path: str, dfm: QlikDFM, write_folder: str) -> str:
+    """
+    Convert snapshot csv file to parquet.
+
+    Raise any exception and abort snapshot processing, all snapshot must be processed.
+
+    :param object: path to snapshot object on S3
+    :param dfm: QlikDFM for snapshot
+    :param write_folder: folder that parquet file will be written to
+
+    :return: path to written parquet file
+    """
+    log = ProcessLog("snapshot_to_parquet", obj_path=obj_path, write_folder=write_folder)
+    try:
+        dfm_dt = datetime.fromisoformat(dfm["fileInfo"]["startWriteTimestamp"])
+        change_seq = dfm_dt.strftime("%Y%m%d%H%M%S").ljust(35, "0")
+        schema = dfm_to_polars_schema(dfm)
+        write_file = os.path.join(write_folder, f"{len(os.listdir(write_folder))}.parquet")
+        with tempfile.TemporaryDirectory() as tmpdir:
+            tmpfile = os.path.join(tmpdir, "temp.csv")
+            download_object(obj_path, tmpfile)
+            pl.scan_csv(
+                tmpfile,
+                schema=schema,
+                has_header=True,
+            ).with_columns(
+                pl.lit(dfm_dt.strftime("%Y")).cast(pl.Int32()).alias("header__year"),
+                pl.lit(dfm_dt.strftime("%m")).cast(pl.Int32()).alias("header__month"),
+                pl.lit(change_seq, dtype=pl.Decimal(35, 0)).alias("header__change_seq"),
+                pl.lit("L").alias("header__change_oper"),
+                pl.lit(dfm_dt).alias("header__timestamp"),
+                pl.lit(obj_path).alias("header__from_csv"),
+            ).sink_parquet(
+                write_file,
+                compression="zstd",
+                compression_level=3,
+                row_group_size=int(1024 * 1024 / (8 * schema.len())),
+            )
+        log.complete()
+
+    except Exception as exception:
+        log.failed(exception)
+        raise exception
+
+    return write_file
+
+
+def change_csv_to_parquet(
```

Comment:
So a naming suggestion here. We should update all 'change' to 'cdc', where it makes sense. Got confused initially about what this function was doing, the 'change_csv' part. I believe it's only dealing with cdc, so suggest to change it to `cdc_to_parquet`.